python/grass/script/raster.py

```diff
@@ -405,3 +406,176 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+        """
+        Initializes the MaskManager.
+
+        :param env: Environment to use. Defaults to modifying os.environ.
+        :param kwargs: Keyword arguments passed to `g.region`
+        """
+        self.env = env if env is not None else os.environ
+        self._original_value = None
+        self.region_name = append_uuid(append_node_pid("region"))
+        self._region_inputs = kwargs or {}
+
+    def set_region(self, **kwargs):
+        """Sets region.
+
+        :param kwargs: Keyword arguments with g.region parameters
+        """
+        run_command("g.region", **kwargs, env=self.env)
+
+    def __enter__(self):
+        """Sets the `WIND_OVERRIDE` environment variable to the generated region name.
+
+        :return: Returns the RegionManager instance.
+        """
+        self._original_value = self.env.get("WIND_OVERRIDE")
+        run_command(
+            "g.region", save=self.region_name, env=self.env, **self._region_inputs
+        )
+        self.env["WIND_OVERRIDE"] = self.region_name
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Restore the previous region state.
+
+        Restores the original value of `WIND_OVERRIDE`.
+
+        :param exc_type: Exception type, if any.
+        :param exc_val: Exception value, if any.
+        :param exc_tb: Traceback, if any.
+        """
+        if self._original_value is not None:
+            self.env["WIND_OVERRIDE"] = self._original_value
+        else:
+            self.env.pop("WIND_OVERRIDE", None)
+            run_command(
+                "g.remove",
+                flags="f",
+                quiet=True,
+                type="region",
+                name=self.region_name,
+                env=self.env,
+            )
+
+
+class RegionManagerEnv:
+    """Context manager for temporarily setting the computational region.
+
+    See :class:`RegionManager`. Unlike :class:`RegionManager`, this class uses
+    `GRASS_REGION` instead of `WIND_OVERRIDE`. The advantage is no files are written to disk.
+    The disadvantage is that simply calling `g.region` within the context will not affect
+    the temporary region, but the global one, which can be confusing.
+
+    Example with explicit region parameters:
+
+    >>> with gs.RegionManagerEnv(n=226000, s=222000, w=634000, e=638000):
+    ...     gs.parse_command("r.univar", map="elevation", format="json")
+
+    Example with set_region():
+
+    >>> with gs.RegionManagerEnv() as manager:
+    ...     manager.set_region(n=226000, s=222000, w=634000, e=638000)
+    ...     gs.parse_command("r.univar", map="elevation", format="json")
+
+    This is identical to:
+
+    >>> with gs.RegionManagerEnv() as manager:
+            manager.env["GRASS_REGION"] = gs.region_env()
+    ...     gs.parse_command("r.univar", map="elevation", format="json")
+
+    .. caution::
+
+        To set region within the context, do not call `g.region`, use `set_region` instead.
```

Comment:
```suggestion
        To set region within the context, do not call ``g.region``, use ``set_region`` instead.
```

I also forgot, reST single back quote is italics (emphasis), double them for code, or use the other syntax for with a column, and all